### PR TITLE
Make temporary folders temporary

### LINKED ISSUE #1565: Tests leave artifact
### Description of feature

When running tests, a hardcoded temporary folder is used and remains on the system even after the tests finished.

The folder is "/tmp/nxf/nf-core/dummy-wf"

### PR DESCRIPTION
**Pull request by fabianegli**
The tests for list left a remnant folder in `/tmp/` named `nxf` which I believe to be an undesired behaviour. 

I also took the opportunity to use pathlib.Path, isort and simplify some of the code in `list.py` (and make it marginally faster).

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

Closes #1565

### CODE REVIEW

**Review by mirpedrol — APPROVED** [review]
Tested the functionality and looks good to me

File: tests/test_list.py [hunk 1]
```diff
@@ -130,7 +136,7 @@ def test_parse_local_workflow_home(self, mock_local_wf, mock_subprocess):
     @mock.patch("git.Repo")
     def test_local_workflow_investigation(self, mock_repo, mock_stat):
         local_wf = nf_core.list.LocalWorkflow("dummy")
-        local_wf.local_path = "/tmp"
+        local_wf.local_path = tmp_nxf
```
[the remark this answers is not in the document]
Comment (mirpedrol) [reply tests/test_list.py:139]: I've checked the test and it seems it's only pulling information from a local dummy git repository, so I think it should be fine to leave local_path as it is, as long as it's not None

**Comment by fabianegli** [comment]
I also sneaked in a Deprecation warning for `invalid escape sequence \[`. A bit off-topic for this PR, but I hope that's fine.

**Comment by ewels** [comment]
Missing changelog, for next time :) (I'll add something now)

**Comment by fabianegli** [comment]
Sorry for that, and thank you for taking care of the changelog.

**Comment by ewels** [comment]
No worries!

**Pull request merged**